packages/browser-destinations/destinations/evolv-ai/src/source/index.ts

```diff
@@ -0,0 +1,70 @@
+import { eventTracking } from './eventTracking'
+import { extractAllocations } from './allocations'
+import { Allocation } from '../types'
+import { prepareEvent } from './events'
+
+async function emit(eventType: string, allocation: Allocation) {
+  if (eventType !== 'confirmed') return
+
+  const event = await prepareEvent(allocation)
+  const data = {
+    nonInteraction: 1,
+    experiment_id: event.group_id,
+    experiment_name: event.experiment_name,
+    variation_id: event.cid,
+    variation_name: `combination ${event.ordinal}`
+  }
+
+  if (window.analytics) {
+    // console.info('sending track for analytics', data)
+    window.analytics
+      .track('Experiment Viewed', data)
+      .then(() => true)
+      .catch(() => true)
+  } else {
+    console.warn('window.analytics not available for track of', data)
+  }
+}
```

Comment:
```suggestion
  if (window.analytics) {
    // console.info('sending track for analytics', data)
    window.analytics
      .track('Experiment Viewed', data)
      .then(() => true)
      .catch(() => true)
  } else {
    console.warn('window.analytics not available for track of', data)
  }
}
```

So are you okay with this functionality not being available for users that load analytics as a module? I'm not sure what the split would be, but it's fair to say that a lot of users wouldn't be able to make use of this source part.

Reply:
Would there be an optional way of doing this that might be more robust?